Loop/Managers/SupportManager.swift

```diff
@@ -0,0 +1,258 @@
+        staticSupportTypesByIdentifier = self.staticSupportTypes.reduce(into: [:]) { (map, type) in
+            map[type.supportIdentifier] = type
+        }
+
+        restoreState()
+
+        ((pluginManager?.availableSupports ?? [SupportUI]()) +
+         (deviceDataManager?.availableSupports ?? [SupportUI]()) +
+         (servicesManager?.availableSupports ?? [SupportUI]()) +
+         self.staticSupportTypes.map { $0.init(rawState: [:]) }.compactMap { $0 })
+            .forEach {
+                addSupport($0)
+            }
+                
+        // Perform a check every foreground entry and every loop
+        NotificationCenter.default.publisher(for: UIApplication.willEnterForegroundNotification)
+            .sink { [weak self] _ in
+                self?.performCheck()
+            }
+            .store(in: &cancellables)
+        
+        NotificationCenter.default.publisher(for: .LoopCompleted)
+            .sink { [weak self] _ in
+                self?.performCheck()
+            }
+            .store(in: &cancellables)
+    }
+}
+
+// MARK: Manage list of Supports
+extension SupportManager {
+    func addSupport(_ support: SupportUI) {
+        supports.mutate {
+            if $0[support.identifier] == nil {
+                $0[support.identifier] = support
+                support.setAlertIssuer(alertIssuer: alertIssuer)
+            }
+        }
+    }
+
+    func restoreSupport(_ support: SupportUI) {
+        addSupport(support)
+    }
+
+    func removeSupport(_ support: SupportUI) {
+        supports.mutate {
+            $0[support.identifier] = nil
+            support.setAlertIssuer(alertIssuer: nil)
+        }
+    }
+    
+    var availableSupports: [SupportUI] {
+        return Array(supports.value.values)
+    } 
+}
+
+// MARK: Version checking
+extension SupportManager {
+    func performCheck() {
+        checkVersion { [self] versionUpdate in
+            notify(versionUpdate)
+        }
+    }
+    
+    private func notify(_ versionUpdate: VersionUpdate) {
+        if versionUpdate.softwareUpdateAvailable {
+            NotificationCenter.default.post(name: .SoftwareUpdateAvailable, object: versionUpdate)
+        }
+    }
+        
+    func checkVersion(completion: @escaping (VersionUpdate) -> Void) {
+        let group = DispatchGroup()
+        var results = [String: Result<VersionUpdate?, Error>]()
+        supports.value.values.forEach { support in
+            group.enter()
+            support.checkVersion(bundleIdentifier: Bundle.main.bundleIdentifier!, currentVersion: Bundle.main.shortVersionString) { result in
+                results[support.identifier] = result
+                group.leave()
+            }
+        }
+        group.notify(queue: DispatchQueue.main) { [weak self] in
+            guard let self = self else { return }
+            self.saveState()
+            let aggregatedResults = self.aggregate(results: results)
```

Comment:
Suggest breaking this tuple apart rather than using `.0` and `.1` which requires extra knowledge to know what each is for. Perhaps `let (identifierWithHighestVersionUpdate, aggregatedVersionUpdate) = self.aggregate(results: results)`? Very easy to see what is what with this change.